pkg/virt-controller/watch/vmi.go

```diff
@@ -2144,13 +2144,14 @@ func (c *VMIController) updateVolumeStatus(vmi *virtv1.VirtualMachineInstance, v
 			}
 			attachmentPod := c.findAttachmentPodByVolumeName(volume.Name, attachmentPods)
 			if attachmentPod == nil {
-				status.HotplugVolume.AttachPodName = ""
-				status.HotplugVolume.AttachPodUID = ""
-				// Pod is gone, or hasn't been created yet, check for the PVC associated with the volume to set phase and message
-				phase, reason, message := c.getVolumePhaseMessageReason(&vmi.Spec.Volumes[i], vmi.Namespace)
-				status.Phase = phase
-				status.Message = message
-				status.Reason = reason
+				if status.Phase != virtv1.VolumeReady {
```

Comment:
sure that's the only value we care about?  also would it make sense to create a helper function instead of explicit check?

Reply:
Yes I am sure, all the other phases the volume is not mounted in the VM and switching phase won't matter. But I can definitely make it a helper. Might explain the intent a little better.